### PR TITLE
Fix plugin use with filters

### PR DESCRIPTION
**Pull request by ikbuibui**
Using filters with plugins was broken and only worked when the filter used by the user was the same as the first filter in the internal ``EligibleFilters`` type list (usually it is the filter ``all``).
In a plugin, when ``meta::ForEach<typename Help::EligibleFilters, plugins::misc::ExecuteIfNameIsEqual<boost::mpl::_1>>{}(m_help->filter.get(m_id), currentStep, bindKernel);`` is called, the filter name string returned from ``m_help->filter.get(m_id)`` is only passed on correctly to the first expansion from the ForEach List. All other expansions were receiving an empty string.
I am not sure why this happens, I suspect that for some reason the compiler is optimizing the string being passed by value into a move and isn't dealing with the ForEach correctly. As a fix the string is now passed as a const reference, which prevents moves. This fixes the problem in my tests. 
Bug found by @JessicaTiebel

### CODE REVIEW

**Review by chillenzer — APPROVED** [review]
Looks okay to me but I don't have enough information to reproduce the bug, so that's a very weak statement here. We should wait for @JessicaTiebel's approval.

**Review by JessicaTiebel — APPROVED** [review]
I tested it with my setup and it works.
Thanks for the quick fix and also update of documentation ✨

**Comment by ikbuibui** [comment]
closing because I cant undo the CI nocompile flag. Opening #4998

**Pull request closed without merging**
